graph/src/blockchain/mod.rs

```diff
@@ -156,15 +162,38 @@ pub trait TriggerFilter<C: Blockchain>: Default + Clone + Send + Sync {
     fn node_capabilities(&self) -> C::NodeCapabilities;
 }
 
-pub trait DataSource<C: Blockchain>: 'static {
+// ETHDEP: `Source` and `Mapping`, at least, are Ethereum-specific.
+pub trait DataSource: 'static + Sized + Send + Sync + Debug {
```

Comment:
Out of curiosity: why the switch from type parameter to associated type?

Reply:
I started getting `type parameter is unconstrained` errors when making types generic over `DataSource`.

Reply:
Also just noticed this `Debug` bound ended up being unnecessary, I'm removing it.